perfkitbenchmarker/benchmarks/silo_benchmark.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs Silo.
+
+Silo is a high performance, scalable in-memory database for modern multicore
+machines
+
+Documentation & code: https://github.com/stephentu/silo
+"""
+
+import logging
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import regex_util
+from perfkitbenchmarker import sample
+from perfkitbenchmarker.packages import silo
+
+FLAGS = flags.FLAGS
+
+BENCHMARK_INFO = {'name': 'silo',
+                  'description': 'Runs Silo',
+                  'scratch_disk': False,
+                  'num_machines': 1}
+
+flags.DEFINE_string('silo_benchmark', 'tpcc', 'benchmark to run with silo')
+
+AGG_THPUT_REGEX = r'(agg_throughput):\s+([\d.]+)\s+([a-z/]+)'
+PER_CORE_THPUT_REGEX = r'(avg_per_core_throughput):\s+([\d.]+)\s+([a-z/]+)'
+LAT_REGEX = r'(avg_latency):\s+([\d.]+)\s+([a-z]+)'
+
+
+def GetInfo():
+  return BENCHMARK_INFO
+
+
+def Prepare(benchmark_spec):
+  """Install Silo on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  vms = benchmark_spec.vms
+  vm = vms[0]
+  logging.info('Silo prepare on %s', vm)
+  vm.Install('silo')
+
+
+def ParseResults(results):
```

Comment:
Parsers always make for good tests.  For example you could added tests to make sure you pull out the right information using this function.  You can try various fake inputs.